runtime/request.go

```diff
@@ -50,3 +53,36 @@ func (p PathParams) Get(name string) string {
 
 	return ""
 }
+
+func (mgr *Manager) CurrentRequest() *Request {
+	req := mgr.rt.Current().Req
+	if req == nil {
+		return &Request{
+			Type:    None,
+			Started: applicationStartTime,
+		}
+	}
+
+	opType := None
+	switch req.Type {
+	case model.RPCCall, model.AuthHandler:
+		opType = APICall
+	case model.PubSubMessage:
```

Comment:
If we're adding `PubSubMessage` here, I think we should add `Topic` and `Subscription` at least to the Request Object. (I would be tempted to think about Message ID, Delivery Attempt and Publish Time as well)